applications/ExaquteSandboxApplication/custom_processes/metrics_divergencefree_process.cpp

```diff
@@ -0,0 +1,385 @@
+            "global_tolerance"                 : 0.1
+        },
+        "echo_level"                          : 0
+    })"
+    );
+
+    ThisParameters.RecursivelyValidateAndAssignDefaults(default_parameters);
+    mMinSize = ThisParameters["minimal_size"].GetDouble();
+    mMaxSize = ThisParameters["maximal_size"].GetDouble();
+    mEchoLevel = ThisParameters["echo_level"].GetInt();
+    mReferenceVariable = ThisParameters["reference_variable_name"].GetString();
+    if (ThisParameters["refinement_strategy"].GetString() == "maximum_strategy") mRefinementStrategy = MaximumStrategy;
+    else if (ThisParameters["refinement_strategy"].GetString() == "mean_distribution_strategy") mRefinementStrategy = MeanDistributionStrategy;
+    else if (ThisParameters["refinement_strategy"].GetString() == "global_tolerance_strategy") mRefinementStrategy = GlobalToleranceStrategy;
+
+    // Mean distribution strategy
+    mMeanStrategyReferenceNorm = ThisParameters["mean_distribution_strategy"]["reference_norm_name"].GetString();
+    mMeanStrategyTargetRefinementCoefficient = ThisParameters["mean_distribution_strategy"]["target_refinement_coefficient"].GetDouble();
+    mMeanStrategyRefinementBound = ThisParameters["mean_distribution_strategy"]["refinement_bound"].GetDouble();
+    mMeanStrategyDivergenceFreeOverAllDomain = 0;
+
+    // Maximum strategy
+    mMaxStrategyTargetRefinementCoefficient = ThisParameters["maximum_strategy"]["target_refinement_coefficient"].GetDouble();
+    mMaxStrategyRefinementCoefficient = ThisParameters["maximum_strategy"]["refinement_coefficient"].GetDouble();
+    mDivergenceFreeMaxValue = -1;
+
+    // Global tolerance strategy
+    mGlobalErrorStrategyGlobalTolerance = ThisParameters["global_tolerance_strategy"]["global_tolerance"].GetDouble();
+    mGlobalErrorStrategyMeshConstant = 1.0;
+
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<SizeType TDim>
+void MetricDivergenceFreeProcess<TDim>::Execute()
+{
+    // 1) Initialize metric
+
+    // Tensor variable definition
+    KRATOS_ERROR_IF_NOT(KratosComponents<Variable<TensorArrayType>>::Has("METRIC_TENSOR_"+std::to_string(TDim)+"D")) << "Import Meshing Application" << std::endl;
+    const Variable<TensorArrayType>& tensor_variable = KratosComponents<Variable<TensorArrayType>>::Get("METRIC_TENSOR_"+std::to_string(TDim)+"D");
+
+    NodesArrayType& nodes_array = mrThisModelPart.Nodes();
+    KRATOS_DEBUG_ERROR_IF(nodes_array.size() == 0) <<  "ERROR:: Empty list of nodes" << std::endl;
+    ElementsArrayType& elements_array = mrThisModelPart.Elements();
+    KRATOS_DEBUG_ERROR_IF(elements_array.size() == 0) <<  "ERROR:: Empty list of elements" << std::endl;
+
+    if (nodes_array.begin()->Has(tensor_variable) == false) {
+        const TensorArrayType zero_array = ZeroVector(3 * (TDim - 1));
+        // Iteration over the nodes
```

Comment:
```suggestion
        // Iteration over the nodes
        const auto it_node_begin = nodes_array.begin();
```

Reply:
Should be r_nodes_array, but that's more difficult to apply

Reply:
What do you mean here? I don't get it

Reply:
Done the `it_node_begin` change.